File: Graveyard/src/de/hannisoft/gaveplan/properties/PointsReader.java
package de.hannisoft.gaveplan.properties;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;
import java.util.Map.Entry;
import java.util.Properties;

import de.hannisoft.gaveplan.model.Point;

public class PointsReader {
    private final double xFactor;
    private final double yFactor;
    private final int xDelta;
    private final int yDelta;
    private double rotation;

    public PointsReader(double xFactor, double yFactor, int xDelta, int yDelta, double rotation) {
        this.xFactor = xFactor;
        this.yFactor = yFactor;
        this.xDelta = xDelta;
        this.yDelta = yDelta;
        this.rotation = rotation;
    }

    public Map<Integer, Point> readPoints() throws IOException {
        Map<Integer, Point> points = new HashMap<>();
        int minX = 0;
        int maxX = 0;
        int minY = 0;
        int maxY = 0;
        Properties props = new Properties();
        props.load(getClass().getClassLoader().getResourceAsStream("plan/points.properties"));
        for (Entry<Object, Object> prop : props.entrySet()) {
            try {
                int pointId = Integer.parseInt(prop.getKey().toString());
                String[] coords = prop.getValue().toString().split("\\,");
                int x = Integer.parseInt(coords[0]);
                int y = Integer.parseInt(coords[1]);
                Point point = new Point(pointId, x, y);
                point.scale(xFactor, yFactor);
                point.move(xDelta, yDelta);
                point.rotate(rotation);
                points.put(pointId, point);
                // System.out.println(point.toString());
                if (minX == 0 || point.getX() < minX) {
                    minX = point.getX();
                }
                if (maxX == 0 || point.getX() > maxX) {
                    maxX = point.getX();
                }
                if (minY == 0 || point.getY() < minY) {
                    minY = point.getY();
                }
                if (maxY == 0 || point.getY() > maxY) {
                    maxY = point.getY();
                }
            } catch (Exception e) {
                System.err.println("Can't convert Property to Point of '" + prop.toString() + "': " + e.getMessage());
                e.printStackTrace();
            }
        }
        System.out.println("x:[" + minX + " " + maxX + "]");
        System.out.println("y:[" + minY + " " + maxY + "]");
        return points;
    }
}

File: src/main/kotlin/de/hannisoft/graveplan/properties/PointsReader.kt
package de.hannisoft.graveplan.properties

import de.hannisoft.graveplan.model.Point
import java.util.*

class PointsReader(private val xFactor: Double, private val yFactor: Double, private val xDelta: Int, private val yDelta: Int, private val rotation: Double) {
    fun readPoints(): MutableMap<Int?, Point?> {
        val points: MutableMap<Int?, Point?> = HashMap<Int?, Point?>()
        var minX = 0
        var maxX = 0
        var minY = 0
        var maxY = 0
        val props = Properties()
        props.load(javaClass.getClassLoader().getResourceAsStream("plan/points.properties"))
        for (prop in props.entries) {
            try {
                val pointId = prop.key.toString().toInt()
                val coords = prop.value.toString().split(",")
                val x = coords[0].toInt()
                val y = coords[1].toInt()
                val point = Point(pointId, x, y)
                point.scale(xFactor, yFactor)
                point.move(xDelta, yDelta)
                point.rotate(rotation)
                points[pointId] = point
                if (minX == 0 || point.x < minX) {
                    minX = point.x
                }
                if (maxX == 0 || point.x > maxX) {
                    maxX = point.x
                }
                if (minY == 0 || point.y < minY) {
                    minY = point.y
                }
                if (maxY == 0 || point.y > maxY) {
                    maxY = point.y
                }
            } catch (e: Exception) {
                System.err.println("Can't convert Property to Point of '$prop': ${e.message}")
                e.printStackTrace()
            }
        }
        println("x:[$minX $maxX]")
        println("y:[$minY $maxY]")
        return points
    }
}